graphite/graphite.go

```diff
@@ -26,6 +26,10 @@ type Series struct {
 
 type DataPoint []json.Number
 
+func (r *Request) CacheKey() string {
+	return fmt.Sprintf("graphite-%d-%d-%s", r.Start.Unix(), r.End.Unix(), r.Targets)
```

Comment:
How about just convert r to JSON, or r.Targets to JSON to avoid potential encoding collisions.